src/pages/Home.tsx

```diff
@@ -0,0 +1,114 @@
+import { useEffect, useRef, useState } from 'react';
+import { Outlet, useNavigate, useParams } from 'react-router';
+import { ITEMS_DISPLAY_LIMIT, searchProducts } from '../services/product';
+import { Product } from '../services/product.types';
+import ProductSearchInput from '../components/Product/Search/Input';
+import ProductSearchResult from '../components/Product/Search/Result';
+import { UNKNOWN_ERROR_MESSAGE } from '../constants/errorMessages';
+import Pagination from '../components/common/Pagination';
+import useSearchQuery from '../hooks/useSearchQuery';
+
+const HomePage = () => {
+  const navigate = useNavigate();
+  const navigateRef = useRef(navigate);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [isPaginationVisible, setIsPaginationVisible] =
+    useState<boolean>(false);
+
+  const [searchQuery, storeSearchQuery] = useSearchQuery();
+  const [totalItems, setTotalItems] = useState<number>(0);
+
+  const [products, setProducts] = useState<Product[]>([]);
+
+  const { page, productId } = useParams<{
+    page?: string;
+    productId?: string;
+  }>();
+  const currentPage = page ? parseInt(page, 10) : 1;
+
+  useEffect(() => {
+    const fetchSearchProducts = async () => {
+      try {
+        setIsLoading(true);
+        setErrorMessage(null);
+
+        const searchResponse = await searchProducts(searchQuery, currentPage);
+        const { skip, total, products } = searchResponse;
+
+        if (skip > total) {
+          navigateRef.current('/search/1');
+          return;
+        }
+
+        setIsPaginationVisible(true);
```

Comment:
i guess that you should not show pagination if there is no result
```suggestion
        setIsPaginationVisible(total > 0);
```